internal/nlretry/handle.go

```diff
@@ -0,0 +1,35 @@
+package nlretry
+
+import (
+	"context"
+
+	"github.com/vishvananda/netlink"
+)
+
+type handle struct {
+	*netlink.Handle
+}
+
+// NewHandle wraps a netlink.Handle to automatically retry netlink functions that
+// raise netlink.ErrDumpInterrupted.
+// Creates an empty netlink.Handle if the handle passed in is nil.
+func NewHandle(h *netlink.Handle) *handle {
+	if h == nil {
+		h = &netlink.Handle{}
+	}
+	return &handle{Handle: h}
+}
+
+func (r *handle) AddrList(link netlink.Link, family int) ([]netlink.Addr, error) {
```

Comment:
Not really a big deal for this one considering the call chain, but since we are just starting this paradigm, we should probably accept the context from the caller here and pass that down so that in the future, other functions where it matters more will follow suit and be able to cancel the retries call if needed.

Reply:
I was split on this as well.

I don't mind updating it to accept the context from the caller instead. Both the wrapped Handler and the LinkQuerier interface are both for internal usage anyway, so there's no reason for us to have to expose the exact same interface as the underlying netlink library.

Reply:
I guess if you don't mind, we might as well do it as I think it will be handy in future extensions to be able to cancel netlink calls if they run too long.

Reply:
Perfecto!